optimize/backend/src/main/java/io/camunda/optimize/license/CamundaLicense.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.license;
+
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
+import org.camunda.bpm.licensecheck.LicenseKeyImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is a duplicate of src/main/java/io/camunda/service/license/CamundaLicense.java
+ *
+ * <p>This class exists because Optimize is not part of the single application, and cannot use any
+ * of the monorepo's modules. Once Optimize is added, the `service` implementation of
+ * `CamundaLicense` can be used, and this Optimize duplicate can be removed
+ */
+public class CamundaLicense {
+
+  public static final String CAMUNDA_LICENSE_ENV_VAR_KEY = "CAMUNDA_LICENSE_KEY";
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaLicense.class);
+  private boolean isValid;
+  private LicenseType licenseType;
+  private boolean isInitialized;
+
+  public CamundaLicense(final String license) {
+    initializeWithLicense(license);
+  }
+
+  public synchronized boolean isValid() {
+    return isValid;
+  }
+
+  public synchronized LicenseType getLicenseType() {
+    return licenseType;
+  }
```

Comment:
out of interest, why do these need to be `synchronized`?

Reply:
This was actually a weird one for me to figure out. I was getting checkstyle errors in the `initializeWithLicense()` method related to deadlocks

I played around with the method to try and solve it, but the issue was that since these two methods are being called by `initializeWithLicense()` they need to be `synchronized` as well